.github/workflows/nixpkgs-vet.yml

```diff
@@ -19,46 +19,34 @@ permissions: {}
 # There is a feature request for suppressing notifications on concurrency-canceled runs: https://github.com/orgs/community/discussions/13015
 
 jobs:
+  get-merge-commit:
+    uses: ./.github/workflows/get-merge-commit.yml
```

Comment:
Do we need a separate job for this? It seems like this can just be a separate step in the first job instead. This way we wouldn't spam the list of CI checks with many of these

Reply:
I just followed https://docs.github.com/en/actions/sharing-automations/reusing-workflows#calling-a-reusable-workflow

But let's try to implement this.

Reply:
@infinisil tried https://github.com/NixOS/nixpkgs/pull/361494/commits/d3446ef3d7072be54facc926568a02830c484823, but seems to fail https://github.com/JohnRTitor/nixpkgs/actions/runs/12145724531/job/33868011344?pr=2

Could you help?

Reply:
You'll need to move it after the checkout, as the message suggests. Otherwise it hasn't cloned the repo.

Reply:
Well we are doing it in the get-merge-commit workflow. https://github.com/JohnRTitor/nixpkgs/blob/d3446ef3d7072be54facc926568a02830c484823/.github/workflows/get-merge-commit.yml#L16 

It works if I just drop https://github.com/NixOS/nixpkgs/pull/361494/commits/d3446ef3d7072be54facc926568a02830c484823

Reply:
On Matrix (since there were problems with GitHub) @JohnRTitor shared [this](https://docs.github.com/en/actions/sharing-automations/reusing-workflows#calling-a-reusable-workflow):

> Unlike when you are using actions within a workflow, you call reusable workflows directly within a job, and not from within job steps.

Reply:
I just found out about "Composite actions" though, which sound like a better fit now! https://docs.github.com/en/actions/sharing-automations/avoiding-duplication#key-differences-between-reusable-workflows-and-composite-actions

Reply:
Looks like that works best with a separate repository though, which we could totally do by [requesting it](https://github.com/NixOS/org/issues), won't be as local anymore though.

Reply:
Composite actions does not need a separate repo, I use them in both in-repo and out-of-repo.

In-repo example https://github.com/azuwis/nix-config/blob/dd9650cce55ca08d1304fb49e850a4ddaeb42f8d/.github/workflows/package.yml#L53

We can totally put them in the `ci/` dir.